yaptide/routes.py

```diff
@@ -110,65 +105,112 @@ def get():
-            user_id, description=f"User with id {user_id} not found.")
-        db.session.delete(user)
-        db.session.commit()
-        return user, 202
-
-    @marshal_with(example_user_fields)
-    def patch(self, user_id):
-        args = example_user_modify_args.parse_args()
-        user = ExampleUserModel.query.get_or_404(
-            user_id, description=f"User with id {user_id} not found.")
-        for key, value in args.items():
-            if value:
-                setattr(user, key, value)
-        db.session.commit()
-        print(user)
-        return user, 202
-
-############################################
+            try:
+                user = UserModel(
+                    login_name=json_data.get('login_name')
+                )
+                user.set_password(json_data.get('password'))
+
+                db.session.add(user)
+                db.session.commit()
+
+                return {
+                    'status': 'SUCCESS',
+                    'message': 'User created'
+                }, api_status.HTTP_201_CREATED
+
+            except Exception:  # skipcq: PYL-W0703
+                return {
+                    'status': 'ERROR',
+                    'message': 'Internal server error'
+                }, api_status.HTTP_500_INTERNAL_SERVER_ERROR
+        else:
+            return {
+                'status': 'ERROR',
+                'message': 'User existing'
+            }, api_status.HTTP_202_ACCEPTED
+
+
+class UserLogIn(Resource):
+    """Class responsible for user log in"""
+
+    class _Schema(Schema):
+        """Class specifies API parameters"""
+
+        login_name = fld.String()
+        password = fld.String()
+
+    @staticmethod
+    def post():
+        """Method returning status of logging in (and token if it was successful)"""
+        try:
+            json_data: dict = UserLogIn._Schema().load(request.get_json(force=True))
+        except ValidationError:
+            return {
+                'status': 'ERROR',
+                'message': 'Wrong data types provided'
+            }, api_status.HTTP_400_BAD_REQUEST
+
+        try:
+            user = db.session.query(UserModel).filter_by(
+                login_name=json_data.get('login_name')).first()
+            if not user:
+                return {
+                    'status': 'ERROR',
+                    'message': 'User not existing'
+                }, api_status.HTTP_404_NOT_FOUND
+            if not user.check_password(password=json_data.get('password')):
+                return {
+                    'status': 'ERROR',
+                    'message': 'Incorrect password'
+                }, api_status.HTTP_401_UNAUTHORIZED
+
+            token = user.encode_auth_token(user_id=user.id)
+            resp = make_response({
+                'status': 'SUCCESS',
+                'message': 'User logged in'
+            }, api_status.HTTP_200_OK)
+            resp.set_cookie('token', token)
```

Comment:
```suggestion
            resp.set_cookie('token', token, httponly=True, samesite='Lax')
```

also expiration date would be useful